x-pack/filebeat/input/awss3/s3_objects.go

```diff
@@ -233,6 +235,65 @@ func (p *s3ObjectProcessor) readJSON(r io.Reader) error {
 	return nil
 }
 
+// readParquet reads a parquet file based on config options and processes it using the parquet reader,
+// decodes it to JSON and sends the decoded data to the readJSON method to process further as JSON
+func (p *s3ObjectProcessor) readParquet(r io.Reader) error {
+	sReader, err := parquet.NewStreamReader(r, &parquet.Config{
+		ProcessParallel: p.readerConfig.ParquetConfig.ProcessParallel,
+		BatchSize:       p.readerConfig.ParquetConfig.BatchSize,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create parquet stream reader: %w", err)
+	}
+	defer sReader.Close()
+
+	for sReader.Next() {
+		data, err := sReader.Record()
+		if err != nil {
+			return fmt.Errorf("failed to read records from parquet record reader: %w", err)
+		}
+		if data != nil {
+			err = p.readParquetJSON(bytes.NewReader(data))
+			if err != nil {
+				return fmt.Errorf("failed to read JSON data from arrow record: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// readParquetJSON uses an array of json.RawMessage to decode parquet json data
+// since the result of readParquet method is always a stringified json array
```

Comment:
```suggestion
// readParquetJSON uses an array of json.RawMessage to decode parquet json data
// since the result of readParquet method is always a stringified json array.
```